### PR TITLE
All block device test with all device names support [wait for utils/5618]

### PR DESCRIPTION
**Pull request by abdhaleegit**
Add support for persistent device name as input disk
    
This code change will allow user to provide not only device
node name /dev/sdx, but any name of the disk by-id, by-path,
by-uuid a mapper device etc which is required in cases where
the disks names are not persistent across os installs etc

### CODE REVIEW

**Comment by AYUSHJAIN951** [comment]
@abdhaleegit Can you please provide run logs

**Comment by PraveenPenguin** [comment]
@abdhaleegit why that much  commits can you please wrap in a valid commit which is aligned

**Comment by PraveenPenguin** [comment]
@abdhaleegit  please resolve the merge conflict and wrap commits in valid commits as this many commits do not make sense
@narasimhan-v as I see it major changes, please have a review

**Comment by abdhaleegit** [comment]
> @abdhaleegit please resolve the merge conflict and wrap commits in valid commits as this many commits do not make sense @narasimhan-v as I see it major changes, please have a review

The changes are undertest and also the reason to have separate commit is many files have different changes for the same functionality support.

will rebase and send the latest code

**Comment by PraveenPenguin** [comment]
> > @abdhaleegit please resolve the merge conflict and wrap commits in valid commits as this many commits do not make sense @narasimhan-v as I see it major changes, please have a review
> 
> The changes are undertest and also the reason to have separate commit is many files have different changes for the same functionality support.
> 
> will rebase and send the latest code

sure but still I feel it should be consolidated into logical commits

**Comment by abdhaleegit** [comment]
@Naresh-ibm Please review the latest code

File: io/disk/lvsetup.py [hunk 1]
```diff
@@ -98,8 +102,8 @@ def setUp(self):
             else:
                 self.lv_size = int(self.lv_size) / 1024 / 1024
 
-        if self.disk:
-            disk_size = lv_utils.get_devices_total_space(self.disk.split())
+        if self.disks:
+            disk_size = lv_utils.get_devices_total_space(self.disks)
```
Comment (Naresh-ibm) [inline io/disk/lvsetup.py:106]: self.disks can be more than one disk based on line #58.
so please take care of this, either you have to pass a single disk or run in a loop

File: io/disk/softwareraid.py [hunk 2]
```diff
@@ -41,30 +41,33 @@ def setUp(self):
...
         """
+        self.disks = []
+        self.spare_disks = []
         smm = SoftwareManager()
         if not smm.check_installed("mdadm"):
             self.log.info("Mdadm must be installed before continuing the test")
             if SoftwareManager().install("mdadm") is False:
                 self.cancel("Unable to install mdadm")
-        disks = self.params.get('disks', default='').strip(" ")
+        disks = self.params.get('disks', default='').split()
```
Comment (Naresh-ibm) [inline io/disk/softwareraid.py:51]: I think you have to add strip too as some time chances of adding space at the end.
please re-think on this

File: io/disk/ssd/blkdiscard.py [hunk 3]
```diff
@@ -48,7 +48,8 @@ def setUp(self):
         smm = SoftwareManager()
         if not smm.check_installed("util-linux"):
             self.cancel("blkdiscard is needed for the test to be run")
-        self.disk = self.params.get('disk', default='/dev/nvme0n1')
+        device = self.params.get('disk', default='/dev/nvme0n1')
```
Comment (Naresh-ibm) [inline io/disk/ssd/blkdiscard.py:51]: @abdhaleegit Please make the `default=None`

File: io/disk/ssd/ezfiotest.py [hunk 4]
```diff
@@ -43,7 +44,8 @@ def setUp(self):
         """
         Build 'fio and ezfio'.
         """
-        self.disk = self.params.get('disk', default='/dev/nvme0n1')
+        device = self.params.get('disk', default='/dev/nvme0n1')
```
Comment (Naresh-ibm) [inline io/disk/ssd/ezfiotest.py:47]: same here, remove the default value

File: io/disk/ssd/nvme_cli_selftests.py [hunk 5]
```diff
@@ -40,9 +41,10 @@ def setUp(self):
         """
         Download 'nvme-cli'.
         """
-        self.device = self.params.get('device', default='nvme0')
-        self.device = "/dev/%s" % self.device
-        self.disk = self.params.get('disk', default='/dev/nvme0n1')
+        nvme_node = self.params.get('device', default='nvme0')
+        self.device = disk.get_absolute_disk_path(nvme_node)
+        device = self.params.get('disk', default='/dev/nvme0n1')
```
Comment (Naresh-ibm) [inline io/disk/ssd/nvme_cli_selftests.py:46]: Please remove the default values

File: io/disk/ssd/nvmetest.py [hunk 6]
```diff
@@ -42,8 +43,8 @@ def setUp(self):
         """
         Build 'nvme-cli' and setup the device.
         """
-        self.device = self.params.get('device', default='nvme0')
-        self.device = "/dev/%s" % self.device
+        nvme_node = self.params.get('device', default='nvme0')
```
Comment (Naresh-ibm) [inline io/disk/ssd/nvmetest.py:46]: remove default value

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: io/disk/lvsetup.py (hunk 1 above)
Comment (abdhaleegit) [reply io/disk/lvsetup.py:106]: its already taken care as discussed offline

**Comment by abdhaleegit** [comment]
As per suggestion splitting common logical changes into single commit and a separate PR

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2606

File: io/disk/ssd/blkdiscard.py (hunk 3 above)
Comment (abdhaleegit) [reply io/disk/ssd/blkdiscard.py:51]: done taken care

**Comment by AYUSHJAIN951** [comment]
> As per suggestion splitting common logical changes into single commit and a separate PR
> 
> #2606

@abdhaleegit 
As #2606 is already merged, is this PR still needed? Just confused if i am missing something

**Comment by abdhaleegit** [comment]
> > As per suggestion splitting common logical changes into single commit and a separate PR
> > #2606
> 
> @abdhaleegit As #2606 is already merged, is this PR still needed? Just confused if i am missing something

@AYUSHJAIN951 some more changes from #2577  is here please have alook .. once the below once are merged. we can close 

https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2631
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2630
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2629
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2628
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2632

**Comment by abdhaleegit** [comment]
> > > As per suggestion splitting common logical changes into single commit and a separate PR
> > > #2606
> > 
> > 
> > @abdhaleegit As #2606 is already merged, is this PR still needed? Just confused if i am missing something
> 
> @AYUSHJAIN951 some more changes from #2577 is here please have alook .. once the below once are merged. we can close
> 
> #2631 #2630 #2629 #2628 #2632

Please ACK above PRs

**Comment by abdhaleegit** [comment]
Closing this as all the changes are already merged for all the above tests

**Pull request closed without merging**